redislock.go

```diff
@@ -57,10 +56,14 @@ func (c *Client) Obtain(key string, ttl time.Duration, opt *Options) (*Lock, err
 		return nil, err
 	}
 
+	// Metadata string is appended to the lock token.
 	value := token + opt.getMetadata()
 	ctx := opt.getContext()
 
-	var backoff *time.Timer
+	// timer can be init first then defer stop
+	backoff := time.NewTimer(opt.getRetryBackoff())
```

Comment:
I don't want to allocate a timer if there is no need for it, hence the timer is allocated below on the first retry attempt. it's not as easy to read, but it's more memory-efficient